challenges/keyof/prompt.md

```diff
@@ -0,0 +1,61 @@
+## The Problem `keyof` solves
+
+The `keyof` operator is used for extracting a union representing the keys (also known as "properties") of a type.
+
+Let's say your audiophile cousin is asking you to digitize her casette tape collection (she's convinced they "sound better" on casette... yawn..).  So you make an object where you count up how many albums you have for each artist:
+
+```ts
+const casettesByArtist = {
+  'Alanis Morissette': 2,
+  'Mariah Carey': 8,
+  'Nirvana': 3,
+  'Oasis': 2,
+  'Radiohead': 3,
+  'No Doubt': 3,
+  'Backstreet Boys': 3,
+  'Spice Girls': 2,
+  'Green Day': 2,
+  'Pearl Jam': 5,
+  'Metallica': 5,
+  'Guns N\' Roses': 2,
+  'U2': 3,
+  'Aerosmith': 4,
+  'R.E.M.': 4,
+  'Blur': 3,
+  'The Smashing Pumpkins': 5,
+  'Britney Spears': 3,
+  'Whitney Houston': 3,
+};
+```
+
+But what if you wanted to extract all these keys as a type?  You _certainly_ would not want to retype all those artist names.
```

Comment:
To make it even clearer, we could even add an example of manually writing an `Artists` union type, specifically showing that we're duplicating the keys, and then showing the solution with `keyof` that avoids this duplication.

Reply:
I gave it a try: I'm not sure what I think but it seems ok?  what do you think:

932440f34d6cdcf903eb199ec8e64da8e3802856